### PR TITLE
spiral/jasonkuhrt

### PR DESCRIPTION
**Pull request by jasonkuhrt**
This solution feels brutish : )

There must be a simpler recursive mathematical calculation that can be used to derive both horizontal and vertical lines. With that in hand it should be possible to draw all the horizontal lines in one pass, then all the vertical lines in one pass.

I'm seeking feedback now because I would like to try and solve at least one more exercise this week.

### CODE REVIEW

File: week4/spiral/jasonkuhrt/Main.hs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import Data.Matrix (Matrix)
+import qualified Data.Matrix as Matrix
+
+
+
+main :: IO ()
+main = do
+  putStrLn "How wide do you want the spiral to be?"
+  putStrLn . render . spiral . (read :: String -> Int) =<< getLine
```
Comment (neongreen) [inline week4/spiral/jasonkuhrt/Main.hs:86]: Any reason for the type annotation here?

**Review by neongreen — APPROVED** [review]
It might've been easier to just add all lines in one go, since for every line you can calculate the position and length, and there are 4 types of lines (top, bottom, left, right). Or maybe not, I haven't actually implemented it. And yeah, you're right, there should be a fully mathematical solution too.

**Comment by neongreen** [comment]
> add all lines in one go

See @balachandranc's solution for a version of that: https://github.com/neongreen/haskell-ex/pull/15

File: week4/spiral/jasonkuhrt/Main.hs (hunk 1 above)
Comment (jasonkuhrt) [reply week4/spiral/jasonkuhrt/Main.hs:86]: Hm! I don't remember now. I think I might have copied something I tested in `ghci` where it was necessary? At any rate I just revised and tested and yup its not needed. Thanks.

**Pull request merged**